tests/integration/src/test/java/org/mule/test/integration/client/MuleClientDispatchExceptionHandlingTestCase.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.integration.client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.mule.module.http.api.HttpConstants.Methods.POST;
+import static org.mule.module.http.api.client.HttpRequestOptionsBuilder.newOptions;
+
+import org.mule.DefaultMuleMessage;
+import org.mule.api.DefaultMuleException;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleEventContext;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.client.LocalMuleClient;
+import org.mule.api.lifecycle.Callable;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.module.http.api.client.HttpRequestOptions;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Tests to validate that MuleClient can be used from JavaComponent/MessageProcessor in order to dispatch an event to
+ * a sub-flow and if the component/processor throws an exception afterwards the main-flow exception strategy handles
+ * it.
+ *
+ * @author guillermo.fernandes@mulesoft.com
```

Comment:
Ok, I'll remove it.

On Fri, Jan 15, 2016 at 2:36 PM, dfeist <notifications@github.com> wrote:

> In
> tests/integration/src/test/java/org/mule/test/integration/client/MuleClientDispatchExceptionHandlingTestCase.java
> <https://github.com/mulesoft/mule/pull/2341#discussion_r49881148>:
>
> > +import org.mule.tck.junit4.rule.DynamicPort;
> > +
> > +import java.util.concurrent.CountDownLatch;
> > +import java.util.concurrent.TimeUnit;
> > +
> > +import org.apache.commons.logging.Log;
> > +import org.apache.commons.logging.LogFactory;
> > +import org.junit.Rule;
> > +import org.junit.Test;
> > +
> > +/**
> > + * Tests to validate that MuleClient can be used from JavaComponent/MessageProcessor in order to dispatch an event to
> > + * a sub-flow and if the component/processor throws an exception afterwards the main-flow exception strategy handles
> > + * it.
> > + *
> > + * @author guillermo.fernandes@mulesoft.com
>
> We don't add author to java files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mulesoft/mule/pull/2341/files#r49881148>.
>



-- 
<http://www.mulesoft.com/>Guillermo Fernandes, Senior Software Engineer Av.
Corrientes 316 Entre Piso, Buenos Aires, C1043AAQ, Argentina
  <https://www.linkedin.com/company/www.mulesoft.com>
<https://twitter.com/mulesoft>   <https://www.facebook.com/MuleSoft>
<https://www.youtube.com/user/mulesoftvids>We're hiring!
<http://www.mulesoft.com/careers>